### PR TITLE
Minor fix update to INSTALL.md

### PR DESCRIPTION
**Pull request by thundergolfer**
## Description of change

Came across this discrepancy when building `main` from source.

## Does this change impact existing behavior?

No breaking change. Docs only.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: doc/INSTALL.md [hunk 1]
```diff
@@ -194,7 +194,7 @@ You can build Mountpoint for Amazon S3 from source. However, binaries built in t
 4. Build Mountpoint for Amazon S3:
    ```
    cd mountpoint-s3
-   cargo build --release
+   cargo +nightly build --release
```
Comment (passaro) [inline doc/INSTALL.md:197]: There should be no need to use the nightly toolchain. We build mountpoint with stable (currently rustc 1.78.0).

Comment (passaro) [reply doc/INSTALL.md:197]: Are you seeing any issues without it?

Comment (thundergolfer) [reply doc/INSTALL.md:197]: Ahh maybe I was using an older stable which only had a feature in nightly. Perhaps this doc could then have a note/hint about not using nightly and upgrading to latest stable.

Comment (thundergolfer) [reply doc/INSTALL.md:197]: Better yet, the project could use `rust-toolchain.toml`

**Pull request closed without merging**